apps/desktop-wallet/src/api/apiDataHooks/token/useFetchToken.ts

```diff
@@ -1,54 +1,29 @@
 import { NFT } from '@alephium/shared'
-import { explorer as e } from '@alephium/web3'
 import { useQuery } from '@tanstack/react-query'
 
-import { DataHook } from '@/api/apiDataHooks/apiDataHooksTypes'
-import useFetchNft from '@/api/apiDataHooks/token/useFetchNft'
-import useFetchFtList from '@/api/apiDataHooks/utils/useFetchFtList'
-import { fungibleTokenMetadataQuery, tokenTypeQuery } from '@/api/queries/tokenQueries'
+import { tokenQuery } from '@/api/queries/tokenQueries'
 import { useAppSelector } from '@/hooks/redux'
 import { selectCurrentlyOnlineNetworkId } from '@/storage/network/networkSelectors'
-import { ListedFT, NonStandardToken, TokenId, UnlistedFT } from '@/types/tokens'
+import { ListedFT, Token, TokenId, UnlistedFT } from '@/types/tokens'
 
-type UseFetchTokenResponse = DataHook<ListedFT | UnlistedFT | NFT | NonStandardToken>
-
-const useFetchToken = (id: TokenId): UseFetchTokenResponse => {
-  const { data: fTList, isLoading: isLoadingFtList } = useFetchFtList()
+const useFetchToken = (id: TokenId) => {
   const networkId = useAppSelector(selectCurrentlyOnlineNetworkId)
 
-  const listedFT = fTList?.find((t) => t.id === id)
-
-  const { data: tokenType, isLoading: isLoadingTokenType } = useQuery(
-    tokenTypeQuery({ id, networkId, skip: isLoadingFtList || !!listedFT })
-  )
-
-  const { data: unlistedFT, isLoading: isLoadingUnlistedFT } = useQuery(
-    fungibleTokenMetadataQuery({
-      id,
-      networkId,
-      skip: isLoadingTokenType || tokenType?.stdInterfaceId !== e.TokenStdInterfaceId.Fungible
-    })
-  )
-
-  const { data: nft, isLoading: isLoadingNft } = useFetchNft({
-    id,
-    skip: isLoadingTokenType || tokenType?.stdInterfaceId !== e.TokenStdInterfaceId.NonFungible
-  })
```

Comment:
This "waterfall query" logic was moved inside a `queryFn`, read below for details